jsonschema/cli.py

```diff
@@ -9,39 +13,134 @@
 from jsonschema import __version__
 from jsonschema._reflect import namedAny
 from jsonschema.validators import validator_for
+from jsonschema.exceptions import SchemaError, ValidationError
+
+
+# ------------------------------------------------------------------------------
+# CLASSES
+# ------------------------------------------------------------------------------
+class CliOutputWriter():
+    PARSING_ERROR_MSG = (
+        "Failed to parse {file_name}. "
+        "Got the following error: {exception}\n"
+    )
+    PRETTY_ERROR_MSG = "===[ERROR]===({object_name})===\n{error}\n"
+    PRETTY_SUCCESS_MSG = "===[SUCCESS]===({object_name})===\n"
+
+    def __init__(
+        self,
+        output_format,
+        oneline_format,
+        stdout=sys.stdout,
+        stderr=sys.stderr,
+    ):
+        self.output_format = output_format
+        self.oneline_format = oneline_format
+        if not self.oneline_format.endswith("\n"):
+            self.oneline_format += "\n"
+        self.stdout = stdout
+        self.stderr = stderr
 
+    def write_parsing_error(self, file_name, exception):
+        if self.output_format == "pretty":
+            msg = self.PRETTY_ERROR_MSG.format(
+                object_name=file_name,
+                error=self.PARSING_ERROR_MSG.format(
+                    file_name=file_name,
+                    exception=exception,
+                ),
+            )
+        elif self.output_format == "plain":
+            msg = self.PARSING_ERROR_MSG.format(
+                file_name=file_name,
+                exception=exception,
+            )
+        else:  # pragma: no cover
+            raise ValueError(
+                "Output mode '{}' is unknown by this function"
+                .format(self.output_format)
+            )
+        self.stderr.write(msg)
 
+    def write_valid_error(self, object_name, error_obj):
```

Comment:
Can we avoid abbreviations? Here `write_valid_error` is particularly confusing, I assume this means `write_validation_error`?

Reply:
Right. This is a confusing abbreviation, I did not notice it myself sorry.